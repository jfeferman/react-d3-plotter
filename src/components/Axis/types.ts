export type AxisProps = {
  orient: string,
  scale: Function,
  translate: string,
}
