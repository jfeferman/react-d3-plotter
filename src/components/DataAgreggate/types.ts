export type DataAgreggateProps = {
  data: Array<Array<number>>;
  xScale: Function;
  yScale: Function;
}
