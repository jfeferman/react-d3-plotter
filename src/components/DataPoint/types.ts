export type DataPointProps = {
  coords: Array<number>,
  xScale: Function,
  yScale: Function,
}
