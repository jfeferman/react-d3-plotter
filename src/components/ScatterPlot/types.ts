export type ScatterPlotProps = {
  data: Array<Array<number>>,
}
