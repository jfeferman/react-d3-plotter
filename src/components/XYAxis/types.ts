import { IPlotSettings } from "../../utils/types";

export type XYAxisProps = {
  xScale: Function;
  yScale: Function;
  settings: IPlotSettings;
}
