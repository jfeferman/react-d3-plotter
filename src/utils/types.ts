export interface IPlotSettings {
  width?: number;
  height?: number;
  padding?: number;
  radius?: number;
}
