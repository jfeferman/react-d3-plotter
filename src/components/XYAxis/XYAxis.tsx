import React from 'react'
import { Axis } from '../Axis'
import { ScatterPlotSettings } from 'components/ScatterPlot'

export type XYAxisProps = {
  xScale: () => number
  yScale: () => number
  settings: ScatterPlotSettings
}

export const XYAxis = ({ xScale, yScale, settings }: XYAxisProps) => {
  const { height, padding } = settings

  return (
    <g className="xy-axis">
      <Axis translate={`translate(0, ${height! - padding!})`} scale={xScale} orientation="bottom" />
      <Axis translate={`translate(${padding}, 0)`} scale={yScale} orientation="left" />
    </g>
  )
}

export default XYAxis
