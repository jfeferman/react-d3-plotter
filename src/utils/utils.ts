import * as d3 from 'd3'
import { ScatterPlotSettings } from 'components/ScatterPlot'

/**
 * Determine the x scale
 * @param data The data set
 * @param settings Plot settings
 */
export const getXScale = (data: number[][], settings: ScatterPlotSettings) => {
  const xMax = d3.max(data, (d: number[]) => d[0])
  return d3
    .scaleLinear()
    .domain([0, xMax])
    .rangeRound([settings.padding, settings.width! - settings.padding! * 2])
}

/**
 * Determine the y scale
 * @param data The data set
 * @param settings Plot settings
 */
export const getYScale = (data: number[][], settings: ScatterPlotSettings) => {
  const yMax = d3.max(data, (d: number[]) => d[1])
  return d3
    .scaleLinear()
    .domain([0, yMax])
    .rangeRound([settings.height! - settings.padding!, settings.padding])
}
