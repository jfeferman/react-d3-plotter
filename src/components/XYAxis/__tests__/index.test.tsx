import React from 'react'
import * as TestRenderer from 'react-test-renderer';
import {
  getXScale,
  getYScale,
} from '../../../utils/testUtil'
import XYAxis from '..'
import { IPlotSettings } from 'utils/types'

describe('components/XYAxis', () => {
  const settings: IPlotSettings = {
    height: 400,
    padding: 8,
  }
  let renderer: TestRenderer.ReactTestRenderer;
  let instance: TestRenderer.ReactTestInstance;

  beforeAll(() => {
    renderer = TestRenderer.create(
      <XYAxis
        settings={settings}
        xScale={getXScale()}
        yScale={getYScale()}
      />,
    )
    instance = renderer.root;
  })

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot()
  })

  it('verifies there are two axis', () => {
    const axis = instance.findByType('g')
    expect(axis.props.children).toHaveLength(2)
    expect(axis.props.children[0].props.orient).toBe('bottom')
    expect(axis.props.children[1].props.orient).toBe('left')
  });
})
