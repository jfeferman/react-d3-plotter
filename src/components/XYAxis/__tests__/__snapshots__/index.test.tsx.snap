// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/XYAxis matches the snapshot of the component 1`] = `
<g
  className="xy-axis"
>
  <g
    className="axis"
    transform="translate(0, 392)"
  />
  <g
    className="axis"
    transform="translate(8, 0)"
  />
</g>
`;
